sdk-cpp/include/do_download_property.h

```diff
@@ -42,56 +46,58 @@ enum class download_property
     integrity_check_info,           // std::string
     integrity_check_mandatory,      // boolean
     total_size_bytes,               // uint64
-    // Fe and beyond
+
+    // For the COM interface, the following properties are available only in Windows 21H1 and beyond
     disallow_on_cellular,           // bool
     http_custom_auth_headers,       // std::string
 };
 
 class download_property_value
 {
 
+// CDownloadImpl is declared as a friend class because it needs to access the platform-specific native value for download_property_value
+// The type of the native value is defined in CDownloadPropertyValueInternal, because DO header files are platform agnostic
+// This is so any user of the SDK does not have to worry about supplying platform specific compile definitions to use the SDK
+friend class details::CDownloadImpl; 
```

Comment:
I'm entirely not sure if the way I implemented this is the best way or if this is a proper usage of friend class.

The other option I thought about was just in time initialization of the native type inside download_property_impl and just scrubbing this class altogether. I couldn't see why a user would want to hold onto the download property value.

If we want to go with that approach, I can just expose a bunch of overloads to SetProperty() accepting these various types, and then create the VARIANT internally in download_impl